base/ca/src/main/java/com/netscape/ca/CAService.java

```diff
@@ -1311,10 +1306,9 @@ public boolean service(IRequest request)
         // request types, not policy.
         // XXX how do we know what to look for in request ?
 
-        if (request.getExtDataInCertInfoArray(IRequest.CERT_INFO) != null)
-            return serviceX509(request);
-        else
-            return false; // Don't know what it is ?????
+        boolean requestContentsAreNull = request.getExtDataInCertInfoArray(IRequest.CERT_INFO) == null;
+        return !requestContentsAreNull && serviceX509(request);
```

Comment:
This return isn't making sense to me based on what this was returning..

Should it be a ternary like:
```
return  (requestContentsAreNull) ? serviceX509(request) : false;
```

Reply:
I'll show my working:

1. Convert to ternary operator expression:
```
return request.getExtDataInCertInfoArray(IRequest.CERT_INFO) != null ? serviceX509(request) : false;
```
2. Invert logic:
```
return request.getExtDataInCertInfoArray(IRequest.CERT_INFO) == null ? false : serviceX509(request);
```
3. Extract `request.getExtDataInCertInfoArray(IRequest.CERT_INFO)` to a variable for readability
```
boolean requestContentsAreNull = request.getExtDataInCertInfoArray(IRequest.CERT_INFO) == null;
return requestContentsAreNull ? false : serviceX509(request);
```
4.  If at least one of the two possible return values is a boolean literal, we can simplify the ternary operator into a simpler logical operator. In this case, we can only return `true` if  `requestContentsAreNull` is `false` and `serviceX509(request)` is `true` - so our expression simplifies to:
 ```
return !requestContentsAreNull && serviceX509(request);
```

Reply:
Very clever, I like it